packages/core/src/project-creation.ts

```diff
@@ -0,0 +1,148 @@
+import { getRecommendedGitIgnore } from './project-structure';
+import { getPackageRoot } from './util/packageInfo';
+import { prompt } from 'enquirer';
+
+function createConfirmationPrompt(name: string, message: string) {
+  return {
+    type: 'confirm',
+    name,
+    message,
+    initial: 'y',
+    default: '(Y/n)',
+    isTrue(input: string | boolean) {
+      if (typeof input === 'string') {
+        return input.toLowerCase() === 'y';
+      }
+
+      return input;
+    },
+    isFalse(input: string | boolean) {
+      if (typeof input === 'string') {
+        return input.toLowerCase() === 'n';
+      }
+
+      return input;
+    },
+    format(): string {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const that = this as any;
+      const value = that.value === true ? 'y' : 'n';
+
+      if (that.state.submitted === true) {
+        return that.styles.submitted(value);
+      }
+
+      return value;
+    },
+  };
+}
+
+export async function confirmProjectCreation(): Promise<{
+  projectRoot: string;
+  shouldAddGitIgnore: boolean;
+}> {
+  return prompt([
+    {
+      name: 'projectRoot',
+      type: 'input',
+      initial: process.cwd(),
+      message: 'Kuai project root:',
+    },
+    createConfirmationPrompt('shouldAddGitIgnore', 'Do you want to add a .gitignore?'),
+  ]);
+}
+
+async function copySampleProject(projectRoot: string) {
+  const packageRoot = getPackageRoot();
+
+  await fsExtra.ensureDir(projectRoot);
+
+  const sampleProjectPath = path.join(packageRoot, 'sample-projects');
+
+  const sampleProjectRootFiles = fsExtra.readdirSync(sampleProjectPath);
+  const existingFiles = sampleProjectRootFiles
+    .map((f) => path.join(projectRoot, f))
+    .filter((f) => fsExtra.pathExistsSync(f))
+    .map((f) => path.relative(process.cwd(), f));
+
+  if (existingFiles.length > 0) {
+    const errorMsg = `We couldn't initialize the sample project because these files already exist: ${existingFiles.join(
+      ', ',
+    )}
+
+Please delete or move them and try again.`;
+    console.log(console.log(errorMsg));
+    process.exit(1);
+  }
+
+  await fsExtra.copy(path.join(packageRoot, 'sample-projects'), projectRoot);
+
+  await fsExtra.remove(path.join(projectRoot, 'LICENSE.md'));
+}
+
+async function addGitIgnore(projectRoot: string) {
+  const gitIgnorePath = path.join(projectRoot, '.gitignore');
+
+  let content = await getRecommendedGitIgnore();
+
+  if (await fsExtra.pathExists(gitIgnorePath)) {
+    const existingContent = await fsExtra.readFile(gitIgnorePath, 'utf-8');
+    content = `${existingContent}
+${content}`;
+  }
+
+  await fsExtra.writeFile(gitIgnorePath, content);
+}
+
+async function createPackageJson() {
+  await fsExtra.writeJson(
+    'package.json',
+    {
+      name: 'kuai-project',
+    },
+    { spaces: 2 },
+  );
+}
+
+export async function createProject(): Promise<void> {
+  if (!(await fsExtra.pathExists('package.json'))) {
+    await createPackageJson();
```

Comment:
Maybe we can copy the `package.json` from sample-projects, and replace the version and name.